aten/src/ATen/native/FractionalMaxPool2d.cpp

```diff
@@ -30,9 +30,15 @@ TORCH_META_FUNC(fractional_max_pool2d) (
   int64_t poolSizeW = pool_size[1];
 
   int64_t ndims = input.ndimension();
-  TORCH_CHECK(input.numel() > 0 && (ndims == 3 || ndims == 4),
-    "non-empty 3D or 4D (batch mode) tensor expected for input, but got: ",
-    ndims);
+  TORCH_CHECK(ndims == 3 || ndims == 4,
+              "fractional_max_pool2d(): Expected non-empty 3D or 4D tensor with optional 0-dim batch size, but got: ",
+              input.sizes());
+  for (int64_t i = 1; i < ndims; ++i) {
+    TORCH_CHECK(input.size(i) > 0,
+                "fractional_max_pool2d(): Expected input to have non-zero dimensions for non-batch dimensions, but got",
```

Comment:
This might be overly pedantic, but I think the terms "dimension" and "size" should be precisely and consistently used to avoid confusion. Each "dimension" has a "size".

"Expected input to have non-zero size for non-batch dimensions, but got:"

Similar phrasing for error messages throughout